app.js

```diff
@@ -1,25 +1,34 @@
-const express = require('express')
-const logger = require('morgan')
-const cors = require('cors')
+import express from 'express';
+import morgan from 'morgan';
+import cors from 'cors';
 
-const contactsRouter = require('./routes/api/contacts')
+import { contactsRouterFunction } from './routes/api/contacts.js';
+import { usersRouterFunction } from './routes/api/users.js';
+import passport from './config/config-passport.js';
+import { auth } from './config/config-passport.js';
 
-const app = express()
+export const makeApp = (usersService, contactsService) => {
+  const app = express();
+  const logger = morgan;
 
-const formatsLogger = app.get('env') === 'development' ? 'dev' : 'short'
+  const formatsLogger = app.get('env') === 'development' ? 'dev' : 'short';
 
-app.use(logger(formatsLogger))
-app.use(cors())
-app.use(express.json())
+  app.use(logger(formatsLogger));
+  app.use(cors());
+  app.use(express.json());
 
-app.use('/api/contacts', contactsRouter)
+  app.use('/api/contacts', auth, contactsRouterFunction(contactsService));
+  app.use('/api/users', usersRouterFunction(usersService));
+  app.use(passport.initialize());
```

Comment:
```
  app.use(passport.initialize());
  app.use('/api/contacts', auth, contactsRouterFunction(contactsService));
  app.use('/api/users', usersRouterFunction(usersService));
```
if you change the order of middlewares, you will have access to `user` data in all endpoints.
just use `req.user` in any endpoint. you should have access to following data:
![image](https://github.com/KrzysztofRozbicki/goit-node-hw-02/assets/142218823/03a50706-6543-485c-b405-8eb46a006c5f)

Reply:
just remember to get rid of password before you pass it to another middleware